RFCs/0025-graph-data-model.md

```diff
@@ -54,51 +63,99 @@ Which is similarly extracted as variables:
 model). RDF blank nodes can be denoted with a label that never conflicts with URI such as **`_:my_blank`** (**`_`** is
 never a valid scheme for a URI).
 
-Even though PartiQL generalizes the graph data model, it is *not required* that a database actually exposes arbitrary
-values at vertex or edge values, this is similar to PartiQL over a relational table, where attributes of a row are
+Even though PartiQL defines a very general graph data model, it is *not required* that a host database actually supports arbitrary
+values at vertices or edges. This is similar to PartiQL over a relational database, where attributes of a row are
 restricted to scalars.
 
-## Data Model Integration with PartiQL Query
+## Addition to the PartiQL data model
 
-The SQL/PGQ specification is currently in progress, but has published elements of their work[^2]. It is important that
-PartiQL aligns to an SQL standard that arises around graph query insofar as it is practically acceptable. Let us
-consider a straw example of what a PartiQL graph query could look like and mean with respect to this data model.
+As outlined above, graphs are introduced as a new category of "native" values in the PartiQL
+data model, on par with scalar literals, structs, lists, and bags:
 
-```sql
-SELECT the_a.name AS src, the_b.name AS dest
-FROM my_graph MATCH (the_a:a) -[the_y:y]-> (the_b:b)
-WHERE the_y.score > 10
+```EBNF
+<partiql value> ::= 
+      <absent value>
+    | <scalar value>
+    | <tuple value>
+    | <collection value>
+    | <graph value>                  //new
 ```
+Consequently, graph values can occur as members of structs, bags, and lists.
+
+## Graph data model
+
+For the graph values themselves, rather than giving a grammar -- as PartiQL specification does for
+other values -- we will define an abstract data model, largely following the one for PGML[^2].
+One of the reasons is that a concrete syntax for graph literals could emerge from ISO/IEC
+standardization, possibly as part of GQL, but no public information is yet available about
+that effort.  When appropriate, graph literals in PartiQL would be covered in a separate RFC.
+
+A *graph* (a PartiQL *graph value*) is a tuple
+
+>  < **Nodes**, **Edges**, **ends**, **labels**, **pay** >
+
+where
+
+- **Nodes** is a finite set of the *nodes* of the graph;
+- **Edges** is a finite set of the *edges* of the graph;
+- **ends** **:** **Edges** --> (**Nodes** *x* **Nodes**) union { {u,v} | u, v in **Nodes** }   
+  is a total function mapping each edge to its *endpoints*,
+  which are an either ordered or an unordered pair of nodes;
+- **labels** **:** (**Nodes** union **Edges**) --> P( *string_value* )
+  is a total function that maps each node and each edge to a set of string labels;
+- **pay** **:** (**Nodes** union **Edges**) --> *partiql_value* \ {`MISSING`}
+  is a total function that maps each node and each edge to its *payload*,
+  which is a PartiQL value that cannot be `MISSING`.
```

Comment:
I think we should call this **payload**, I was a little puzzled by the shorthand at first